### PR TITLE
CBL-4230: Analysis Warnings (10/13) Networking/HTTP + Networking/WebSockets

### PR DESCRIPTION
**Pull request by callumbirks**
The only major change is improving `WebSocketProtocol.hh` by replacing `char` with `std::byte`, and a few other type changes in the file. The underlying type of `char` and `std::byte` is essentially the same, but `std::byte` is both more type-safe and also a lot clearer on what it's representing.

### CODE REVIEW

File: Networking/WebSockets/WebSocketProtocol.hh [hunk 1]
```diff
@@ -337,25 +332,28 @@ namespace uWS {
...
-                char* start = dst + headerLength;
-                char* stop  = start + length;
-                int   i     = 0;
-                while ( start != stop ) { (*start++) ^= mask[i++ % 4]; }
+                uint32_t mask_i32 = *reinterpret_cast<uint32_t*>(mask);
+
+                auto*     start = reinterpret_cast<uint32_t*>(dst + headerLength);
+                uint32_t* end   = start + length / 4;
+
+                while ( start != end ) { *start++ ^= mask_i32; }
```
Comment (pasin) [inline Networking/WebSockets/WebSocketProtocol.hh:340]: Additional logic? Is this needed?

File: Networking/WebSockets/WebSocketProtocol.hh [hunk 2]
```diff
@@ -303,28 +298,28 @@ namespace uWS {
...
             }
 
             int flags = 0;
-            dst[0]    = char((flags & SND_NO_FIN ? 0 : 128) | (compressed ? SND_COMPRESSED : 0));
-            if ( !(flags & SND_CONTINUATION) ) { dst[0] |= opCode; }
+            dst[0]    = (std::byte)((flags & SND_NO_FIN ? 0 : 128) | (compressed ? SND_COMPRESSED : 0));
+            if ( !(flags & SND_CONTINUATION) ) { dst[0] |= (std::byte)opCode; }
 
-            char mask[4];
+            std::byte mask[4];
```
Comment (borrrden) [inline Networking/WebSockets/WebSocketProtocol.hh:322]: This looks weird before, and still weird now....does anything actually set these values or are they just...garbage?

File: Networking/WebSockets/WebSocketProtocol.hh [hunk 3]
```diff
@@ -303,28 +298,28 @@ namespace uWS {
             return 0;
         }
 
-        static inline size_t formatMessage(char* dst, const char* src, size_t length, OpCode opCode,
+        static inline size_t formatMessage(std::byte* dst, const char* src, size_t length, OpCode opCode,
```
Comment (borrrden) [inline Networking/WebSockets/WebSocketProtocol.hh:301]: If you are changing to std::byte, why is src still const char*?

File: Networking/WebSockets/WebSocketProtocol.hh (hunk 1 above)
Comment (borrrden) [reply Networking/WebSockets/WebSocketProtocol.hh:340]: It seems to be addressing the comment "This is not optimal" by making it a bit more optimal.

**Review by borrrden — APPROVED** [review]

**Review by jianminzhao — APPROVED** [review]

**Pull request merged**